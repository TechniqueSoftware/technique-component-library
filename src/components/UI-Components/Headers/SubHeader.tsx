import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import * as React from 'react';

export interface SubHeaderProps {
  subheader: React.ReactNode | string;
  subtext?: React.ReactNode | string;
  gutterBottom?: boolean;
  rightComponent?: React.ReactNode;
}

export function SubHeader(props: SubHeaderProps) {
  const {
    subheader,
    subtext,
    gutterBottom = true
  } = props;

  return <Box mb={gutterBottom ? 2 : 0}>
    <Typography variant="body1" component="h2" gutterBottom={true}>
      <b>{subheader}</b>
    </Typography>
    <Typography variant="body1" gutterBottom={true}>
      {subtext}
    </Typography>
  </Box>;
}
