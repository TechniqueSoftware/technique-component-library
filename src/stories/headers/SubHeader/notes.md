# SubHeader
This component provides an abstraction for when we want to have a subheader to divide sections. It provides an optional
subtext parameter and enforces the use of the divider line. 

## JIRA
Related jira ticket: https://clubos.atlassian.net/browse/CR-793
