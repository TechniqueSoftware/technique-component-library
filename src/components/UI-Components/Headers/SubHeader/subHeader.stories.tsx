import * as React from 'react';
import { createCUICStoriesOf } from '../../utils/utils';
import { MODULE_NAME } from '../../constants';
import clubOsTheme from '../../storyThemes/index';
import { MuiThemeProvider } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import { SubHeader, SubHeaderProps } from './SubHeader';
import { text, boolean } from '@storybook/addon-knobs';
import Typography from '@material-ui/core/Typography/Typography';
import HistoryIcon from '@material-ui/icons/History';

function render() {

  const reInterestedSubHeaderProps: SubHeaderProps = {
    subheader: text('subheader', 'Re-Interested Threshold'),
    subtext: text('subtext', 'Re-Interested Leads Threshold setting must be between 0-5000'),
    gutterBottom: boolean('gutterBottom', true),
    rightComponent: <HistoryIcon />
  };

  return (
    <div>
      <MuiThemeProvider theme={clubOsTheme}>
        <Box p={2}>
          <SubHeader {...reInterestedSubHeaderProps} />
          <Typography paragraph={true}>
            This text is here only to illustrate the margin below this component.
          </Typography>
        </Box>
      </MuiThemeProvider>
    </div>);
}

createCUICStoriesOf(MODULE_NAME.HEADERS, module)
  .addCUICStory({
    render,
    name: 'SubHeader',
    notes: require('./notes.md').default
  });
