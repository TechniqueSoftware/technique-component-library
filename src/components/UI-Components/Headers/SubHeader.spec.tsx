import * as React from 'react';
import { SubHeader, SubHeaderProps } from './SubHeader';
import { shallow } from 'enzyme';

describe('SubHeader', () => {
  test('should render with minimal props', () => {
    const props: SubHeaderProps = { subheader: 'mockHeader' };
    const shallowWrapper = shallow(<SubHeader {...props} />);
    expect(shallowWrapper.debug()).toMatchSnapshot();
  });

  test('should render with right icon', () => {
    const props: SubHeaderProps = {
      subheader: 'mockHeader',
      rightComponent:<div>Right Icon</div>
    };
    const shallowWrapper = shallow(<SubHeader {...props} />);
    expect(shallowWrapper.debug()).toMatchSnapshot();
  });

  test('should render a subtitle when provider with one', () => {
    const props: SubHeaderProps = {
      subheader: 'mockHeader',
      subtext: 'mockSubTitle'
    };
    const shallowWrapper = shallow(<SubHeader {...props} />);
    const subTitle = shallowWrapper.findWhere((node) => node.text().indexOf(props.subtext as string) > 0);
    expect(subTitle.length).toBe(1);
    expect(shallowWrapper.debug()).toMatchSnapshot();
  });

  test('should remove margin when gutterBottom is false', () => {
    const props: SubHeaderProps = {
      subheader: 'mockHeader',
      subtext: 'mockSubTitle',
      gutterBottom: false
    };
    const shallowWrapper = shallow(<SubHeader {...props} />);
    const subTitle = shallowWrapper.findWhere((node) => node.text().indexOf(props.subtext as string) > 0);
    expect(shallowWrapper.debug()).toMatchSnapshot();
    expect(subTitle.props().mb).toBe(0);
  });
});
