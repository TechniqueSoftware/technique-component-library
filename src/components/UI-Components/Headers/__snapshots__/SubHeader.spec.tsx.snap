// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader should remove margin when gutterBottom is false 1`] = `
"<Styled(MuiBox) mb={0}>
  <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" component=\\"h2\\" gutterBottom={true}>
    <b>
      mockHeader
    </b>
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" gutterBottom={true}>
    mockSubTitle
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>"
`;

exports[`SubHeader should render a subtitle when provider with one 1`] = `
"<Styled(MuiBox) mb={2}>
  <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" component=\\"h2\\" gutterBottom={true}>
    <b>
      mockHeader
    </b>
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" gutterBottom={true}>
    mockSubTitle
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>"
`;

exports[`SubHeader should render with minimal props 1`] = `
"<Styled(MuiBox) mb={2}>
  <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" component=\\"h2\\" gutterBottom={true}>
    <b>
      mockHeader
    </b>
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" gutterBottom={true} />
</Styled(MuiBox)>"
`;

exports[`SubHeader should render with right icon 1`] = `
"<Styled(MuiBox) mb={2}>
  <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" component=\\"h2\\" gutterBottom={true}>
    <b>
      mockHeader
    </b>
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" gutterBottom={true} />
</Styled(MuiBox)>"
`;
